engine/runtime/src/main/scala/org/enso/interpreter/runtime/IrToTruffle.scala

```diff
@@ -438,191 +438,203 @@ class IrToTruffle(
           dataflowInfo
         )
 
-        val function = methodDef.body match {
```

Comment:
This `methodDef.body` was materializing body of 2500 of functions prior to 177c44e303b9c9a4e8f949c2e1370c852921b1cc - to be effective we have to materialize only functions that are being executed, e.g. avoid `isBuiltinMethod(methodDef.body.body)` check done while processing the module.